packages/server/graphql/types/Organization.ts

```diff
@@ -106,6 +107,14 @@ const Organization: GraphQLObjectType<any, GQLContext> = new GraphQLObjectType<a
           .sort((a, b) => a.name.localeCompare(b.name))
       }
     },
+    isTeamLead: {
+      type: new GraphQLNonNull(GraphQLBoolean),
+      description: 'true if the viewer is a team lead for a team in the organization',
+      resolve: async (_, _args: unknown, {authToken, dataLoader}) => {
+        const viewerId = getUserId(authToken)
+        return authToken.tms.some((teamId) => isTeamLead(viewerId, teamId, dataLoader))
```

Comment:
-1 Did not test, but Array.some would call isTeamLead synchronous which would always be a truthy result (a Promise)